sdk/core/azure-core/inc/azure/core/internal/contract.hpp

```diff
@@ -3,7 +3,7 @@
 
 #pragma once
 
-#ifdef NO_CONTRACTS_CHECKING
+#if defined(NO_CONTRACTS_CHECKING)
 #define AZ_CONTRACT(condition, error)
```

Comment:
Lol, so we do have such macros in our code. We can use these for preconditions :)

Doesn't look like these are being used anywhere in the SDK. @RickWinter, @JeffreyRichter - why were they initially added, here: https://github.com/Azure/azure-sdk-for-cpp/pull/33?

Reply:
This is not good. It violates our guidelines. https://azure.github.io/azure-sdk/cpp_design.html#macros

Reply:
Aren't macros allowed for such cases, following:
> Emission or supression of debugging asserts.

Reply:
I believe that's referring to things like `NDEBUG`

https://stackoverflow.com/questions/2290509/debug-vs-ndebug

Reply:
I have no memory of these things. I'd be OK with adding contract checking to our SDK code that gets compiled away in a non-debug build if others want this.